src/ConversionContext.h

```diff
@@ -8,42 +8,42 @@
 #ifndef DRAFTER_CONVERSIONCONTEXT_H
 #define DRAFTER_CONVERSIONCONTEXT_H
 
+#include <boost/container/vector.hpp>
+
 #include "refract/Registry.h"
-#include "snowcrash.h"
 #include "SourceMapUtils.h"
 
-namespace drafter
+namespace snowcrash
 {
+    struct SourceAnnotation;
+}
 
-    struct WrapperOptions;
-
+namespace drafter
+{
     class ConversionContext
     {
-        refract::Registry registry;
-        const NewLinesIndex newLinesIndex;
-
     public:
-        ConversionContext(const char* source, const WrapperOptions& options);
+        using Warnings = boost::container::vector<snowcrash::SourceAnnotation>;
```

Comment:
Is there some specific reason to prefer  boost vector?

Reply:
The standard doesn't say whether in `std::vector<T,...>` the type `T` must be defined or not. I think specifically Microsoft's implentation requires a definition. `boost::container::vector<T,...>` doesn't depend on the definition of T.
In my refactoring, I moved the snowcrash include out of the header, and forward declared `snowcrash::SourceAnnotation`.